src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -162,7 +163,7 @@ Copyright (c) .NET Foundation. All rights reserved.
                             ResolveRuntimePackAssets;
                             _ComputePackageReferencePublish"
           BeforeTargets="CopyFilesToOutputDirectory"
-          Condition="'$(GenerateDependencyFile)' == 'true'"
+          Condition="'$(GenerateDependencyFile)' == 'true' and ('$(TargetExt)' == 'exe' or '$(Language)' != 'C++')"
```

Comment:
just '$(Language)' != 'C++'

Reply:
also disbale https://github.com/dotnet/sdk/blob/4094dca1091f32f16b5756ef14e6e1f99d67bee3/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L68

may need to check extension when c++ is not set

Reply:
double check what need to be done if we want to support exe, maybe pass an empty is better

Reply:
https://github.com/dotnet/sdk/blob/74b343a9f4db2924a37e78de817b2d314f812c16/src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs#L111 is the reason GenerateDependencyFile fails. If we have to support exe one day. We could return `new Lockfile()` on this line

Reply:
by the time setting `GenerateDependencyFile`, language is available